internal/commands/launch.go

```diff
@@ -64,60 +77,62 @@ func (l *LaunchCommand) Execute() error {
 
-	// 4. fetch grant token for launcher
+	for {
+		// 4. fetch grant token for launcher
 
-	launcherGrantToken, err := auth.FetchGrantToken(n8nURI, token)
-	if err != nil {
-		return fmt.Errorf("failed to fetch grant token for launcher: %w", err)
-	}
+		launcherGrantToken, err := auth.FetchGrantToken(n8nURI, token)
+		if err != nil {
+			return fmt.Errorf("failed to fetch grant token for launcher: %w", err)
+		}
 
-	runnerEnv = append(runnerEnv, fmt.Sprintf("N8N_RUNNERS_GRANT_TOKEN=%s", launcherGrantToken))
+		runnerEnv = append(runnerEnv, fmt.Sprintf("N8N_RUNNERS_GRANT_TOKEN=%s", launcherGrantToken))
 
-	logs.Logger.Println("Fetched grant token for launcher")
+		logs.Logger.Println("Fetched grant token for launcher")
 
-	// 5. connect to main and wait for task offer to be accepted
+		// 5. connect to main and wait for task offer to be accepted
 
-	handshakeCfg := auth.HandshakeConfig{
-		TaskType:   l.RunnerType,
-		N8nURI:     n8nURI,
-		GrantToken: launcherGrantToken,
-	}
+		handshakeCfg := auth.HandshakeConfig{
+			TaskType:   l.RunnerType,
+			N8nURI:     n8nURI,
+			GrantToken: launcherGrantToken,
+		}
 
-	if err := auth.Handshake(handshakeCfg); err != nil {
-		return fmt.Errorf("handshake failed: %w", err)
-	}
+		if err := auth.Handshake(handshakeCfg); err != nil {
+			return fmt.Errorf("handshake failed: %w", err)
+		}
 
-	// 6. fetch grant token for runner
+		// 6. fetch grant token for runner
 
-	runnerGrantToken, err := auth.FetchGrantToken(n8nURI, token)
-	if err != nil {
-		return fmt.Errorf("failed to fetch grant token for runner: %w", err)
-	}
+		runnerGrantToken, err := auth.FetchGrantToken(n8nURI, token)
+		if err != nil {
+			return fmt.Errorf("failed to fetch grant token for runner: %w", err)
+		}
 
-	runnerEnv = append(runnerEnv, fmt.Sprintf("N8N_RUNNERS_GRANT_TOKEN=%s", runnerGrantToken))
+		runnerEnv = append(runnerEnv, fmt.Sprintf("N8N_RUNNERS_GRANT_TOKEN=%s", runnerGrantToken))
 
-	// 7. launch runner
+		// 7. launch runner
 
-	logs.Logger.Println("Task ready for pickup, launching runner...")
-	logs.Logger.Printf("Command: %s", runnerConfig.Command)
-	logs.Logger.Printf("Args: %v", runnerConfig.Args)
-	logs.Logger.Printf("Env vars: %v", env.Keys(runnerEnv))
+		logs.Logger.Println("Task ready for pickup, launching runner...")
+		logs.Logger.Printf("Command: %s", runnerConfig.Command)
+		logs.Logger.Printf("Args: %v", runnerConfig.Args)
+		logs.Logger.Printf("Env vars: %v", env.Keys(runnerEnv))
 
-	cmd := exec.Command(runnerConfig.Command, runnerConfig.Args...)
-	cmd.Env = runnerEnv
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
+		cmd := exec.Command(runnerConfig.Command, runnerConfig.Args...)
+		cmd.Env = runnerEnv
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
 
-	err = cmd.Run()
-	if err != nil {
-		return fmt.Errorf("failed to launch task runner: %w", err)
-	}
+		err = cmd.Run()
+		if err != nil {
+			return fmt.Errorf("task runner process failed: %w", err)
+		}
```

Comment:
Actually we want to continue looping in this case. For example if the runner's node process OOMs, we don't want to stop the launcher, but instead wait for next task and start the runner again when there's work. The reason for this is that we realized that if the launcher container goes offline, the entire pod becomes unreachable, and the main app won't be available either. Hence we want to keep the launcher running as long as possible.